lib/rucio/web/ui/static/rucio.js

```diff
@@ -506,6 +510,7 @@ RucioClient.prototype.create_rule = function(options) {
                  data: JSON.stringify({
                      'dids': options.dids,
                      'account': this.account,
+                     'vo': this.vo,
```

Comment:
Why do we need to pass the VO as part of the data? I don't _fully_ understand what's going on here, but it looks like we've added the VO to the headers at the start of the file (which I'd hope would allow you to get the VO from the `ctx.env.get()` function), and at the [REST layer for adding a rule](https://github.com/stfc/rucio/blob/feature-2635-multiVO_functionality-subissue-8/lib/rucio/web/rest/webpy/v1/rule.py#L226) we don't extract VO from the dictionary of json data.
Same applies to lifetime exception below, I can't actually find the code for `/traces/`, but I'd hope that also gets VO from `ctx…` rather than json data.

Reply:
In the case of traces, the headers that are passed are not the headers of the RucioClient, so the VO is 'needed' there (quotes because I'm not sure it's actually consumed) but in the other two cases you're right that the VO should be passed in the headers. I had added this in because of the `'account': this.account` line above and the general rule that where there's an account, there should be a VO - but I'll have another look at this because yeah it's probably not necessary.